### ISSUE TITLE
[New Data Source]: `aws_resourceexplorer2_search` 

### ISSUE THREAD

**Issue by breathingdust** [hashicorp/terraform-provider-aws#36033]
### Description

Provide a data source allowing a configured resource explorer view to be queried. Investigate whether we would need additional resources to enable multi-account search.

### Requested Resource(s) and/or Data Source(s)

* aws_resourceexplorer2_search

### Potential Terraform Configuration

```terraform
data "aws_resourceexplorer2_search" "example" {
  view_arn = "exampleview"
  
  query_string = "*"
}
```

### References

- https://docs.aws.amazon.com/resource-explorer/latest/apireference/Welcome.html
- https://docs.aws.amazon.com/resource-explorer/latest/apireference/API_Search.html
- https://docs.aws.amazon.com/resource-explorer/latest/userguide/manage-service-multi-account.html

### Would you like to implement a fix?

None

**Comment by ewbankkit** [comment]
Relates https://github.com/hashicorp/terraform-provider-aws/issues/27717.

**Comment by jar-b** [comment]
closed by #36560
